modules/Profile/components/TeacherProfile/TeacherProfile.tsx

```diff
@@ -0,0 +1,81 @@
+import { Grid, Table, Title, Box } from "@mantine/core";
+import { notifications } from "@mantine/notifications";
+
+import { ITeacherProfileProps } from "./TeacherProfile.interface";
+
+const TeacherProfile: React.FC<ITeacherProfileProps> = ({ user }) => {
+  if (!user || !user.teacher) {
+    notifications.show({
+      title: "Profile Data Unavailable",
+      message: "No teacher profile data is available at the moment.",
+      color: "red",
+    });
+  }
+
+  if (!user || !user.teacher) return null;
```

Comment:
Same comment as above

Reply:
1. Should the user be optional if you are checking for the user in ProfileContainer anyway?
2. Aren't we already checking if user is a teacher in ProfileContainer before rendering this component?